plugins/MantisMarkdown/MantisMarkdown.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * MantisBT - A PHP based bugtracking system
+ *
+ * MantisBT is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * MantisBT is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @copyright Copyright 2002  MantisBT Team - mantisbt-dev@lists.sourceforge.net
+ */
+
+require_api( 'mention_api.php' );
+
+/**
+ * Mantis Core Formatting plugin
+ */
+class MantisMarkdownPlugin extends MantisPlugin {
+	
+	/**
+	 * Initialized any needed methods, api's, etc... 
+	 *
+	 * Make sure to turn off Text Processing, causing markdown to not render properly
+	 * @return void
+	 */	
+	function init() {
+
+		if ( ON == config_get( 'plugin_MantisCoreFormatting_process_text' ) ) {
+			config_set( 'plugin_MantisCoreFormatting_process_text', OFF );
+		}
+
+		if ( ON == config_get( 'plugin_MantisCoreFormatting_process_urls' ) ) {
+			config_set( 'plugin_MantisCoreFormatting_process_urls', OFF );
+		}
+			
+		#images can be referenced from internet.
+		http_csp_add( 'img-src', "*" );
```

Comment:
You should call this method from a method that gets triggered from EVENT_CORE_HEADERS event.  Check MantisGraph plugin for an example.